src/profile/profiles.js

```diff
@@ -0,0 +1,100 @@
+import {ObjectID} from 'mongodb';
+import {Promise} from 'es6-promise';
+import db from '../db'
+import PassportSessions from '../dbmodels/session.js'
+import WrioUsers from '../dbmodels/wriouser.js'
+import {dumpError} from '../utils.js'
+import request from 'superagent'
+import nconf from '../wrio_nconf.js'
+
+function generateWrioID() {
+    var min = 100000000000;
+    var max = 999999999999;
+    var id = Math.floor(Math.random() * (max - min) + min);
+    return id;
+}
+
+/* Helper func */
+
+var checkIdExists = async (wrioID) => {
+    var wrioUsers = new WrioUsers();
+    try {
+        await wrioUsers.getByWrioID(wrioID);
+        return true;
+    } catch (e) {
+        return false;
+    }
+};
+
+/* Creation of temporary wrioID */
+
+var storageCreateTempAccount = async (session) => {
+    var wrioUsers = new WrioUsers();
+    var id = generateWrioID();
+    if (await checkIdExists(id)) {
+        return await storageCreateTempAccount(session); // call ourselves until we find unique ID
+    } else {
+        var profile = {
+            wrioID: id.toString(),
+            temporary: true,
+            created: new Date().getTime()
+        };
+        console.log("Creating new user profile",profile);
+        var user = await wrioUsers.create(profile);
+        return user;
+    }
+};
+
+/* Save template records for the user to S3 */
+
+var requestSave = (sid) => {
+        console.log(sid); // TODO: change to safer auth method
+        let api_request = "http://storage"+nconf.get('server:workdomain')+'/api/save_templates?sid='+sid;
```

Comment:
won't work, it's intersite request, shoud be strictly https after we transit to https. Until then it's security hole.